### ISSUE TITLE
 Visualizer window doesn't open in latest Visual Studio (BinaryFormatter not supported)

### ISSUE THREAD

**Issue by r-pankevicius** [agileobjects/ReadableExpressions#125]
Microsoft Visual Studio Professional 2022 (64-bit) - Current
Version 17.5.5

ReadableExpressions.Visualizers 4.5.0

When trying to view generated Automapper expression tree ([example](https://docs.automapper.org/en/stable/Understanding-your-mapping.html)) in the visualizer this error is shown, Visualizer window doesn't open.
![image](https://github.com/agileobjects/ReadableExpressions/assets/1467109/0dc5c8b0-c22b-42f0-bc83-621968f3d2aa)

May be related to this breaking change: [BinaryFormatter serialization methods are obsolete and prohibited in ASP.NET apps](https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/5.0/binaryformatter-serialization-obsolete).

**Comment by sdanyliv** [comment]
Add to your `csproj` `EnableUnsafeBinaryFormatterSerialization`
```xml
<PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    ....
    <EnableUnsafeBinaryFormatterSerialization>true</EnableUnsafeBinaryFormatterSerialization>
</PropertyGroup>
```

**Comment by sdanyliv** [comment]
Well, it is something another, Embedded Web Browser crashed.

**Comment by r-pankevicius** [comment]
Well, not so long time ago Microsoft changed default browser from IE to Edge... Could be the reason...

**Comment by SteveWilkes** [comment]
Hi,

Thanks for helping, @sdanyliv - the original issue is [this one](https://github.com/agileobjects/ReadableExpressions#aspnet-core-5-known-issue), and the work-around XML snippet should only need to be added to entry-point projects (for example web apps, console apps, Azure functions, etc).

The follow-up crash issue is a COM call failure in Visual Studio's debugger visualizer system - this can often be fixed by closing + reopening VS, and/or by running VS as an administrator.

**Comment by SteveWilkes** [comment]
So it turns out as of VS 17.6, Microsoft deprecated `BinaryFormatter` serialization and replaced it with an option to use JSON serialization - I've released v4.6 of the visualizers [to VS Marketplace](https://marketplace.visualstudio.com/items?itemName=vs-publisher-1232914.ReadableExpressionsVisualizers), which includes a VS 17.6+ version of the visualizers which serializes using JSON. Thanks!
